src/world/action/jump.c

```diff
@@ -1,15 +1,250 @@
 #include "common.h"
 
-INCLUDE_ASM(void, "world/action/jump", func_802B6000_E24040, void);
+extern void phys_adjust_cam_on_landing();
+extern void sfx_play_sound_at_player(s32 soundID, s32 arg0);
+extern void suggest_player_anim_clearUnkFlag(s32 arg0);
+extern void phys_init_integrator_for_current_state();
+extern s32 check_input_jump();
+extern s32 player_raycast_below_cam_relative(PlayerStatus*, f32*, f32*, f32*, f32*, f32*, f32*, f32*, f32*);
 
-INCLUDE_ASM(void, "world/action/jump", func_802B60B4_E240F4, void);
+extern f32 D_8010C960;
+extern f32 D_8010C97C;
 
-INCLUDE_ASM(void, "world/action/jump", func_802B6198_E241D8, void);
+void func_802B6000_E24040() {
+    s32 temp_v1;
+    s32 phi_a0;
+    PlayerStatus* playerStatus = &gPlayerStatus;
```

Comment:
We typically put the temps for globals before everything else in the function. So the playerStatus and collisionStatus can go before the temp / phi. Also, you may be able to initialize collisionStatus when you declare it as well.